ktor-network/common/src/io/ktor/network/selector/SelectorManager.kt

```diff
@@ -0,0 +1,17 @@
+package io.ktor.network.selector
+
+import io.ktor.utils.io.core.*
+import kotlinx.coroutines.*
+import kotlin.coroutines.*
+
+/**
+ * Creates the selector manager for current platform.
+ */
+@Suppress("FunctionName")
+public expect fun SelectorManager(
+    dispatcher: CoroutineContext = EmptyCoroutineContext
+): SelectorManager
+
+public expect interface SelectorManager : CoroutineScope, Closeable {
```

Comment:
What it the reason at all to provide this empty `SelectorManager` in common API, when f.e. nodejs sockets, can be implemented without it, as there is only single global event loop there?

Reply:
We need to support the platforms without default event loop, so we need a "manager". (maybe not `selector manager`. Actually I'm just following the origin name here since I didn't define any member related to `Select` on common)

For those platforms that only single global event loop is avaliable, it makes no sence itself. But if we want to write codes on common, we need a placeholder or we'll be hard on commonizing the user code.

Reply:
my question is more about: why at all expose an interface in common, which has no sense on JS?
We can create another `aSocket` functions, which will not use `selector manager` in multiplatform code f.e.

Overall, I don't understand, why to start with this?
Exposing *Socket classes - ok
Exposing Select* classes - not ok, as for now, it's more of an internal detail, rather then something, that could be used outside of ktor-network independent from ktor-network itself.

IMO: better to start trying to implement nodejs/mingw sockets, and see, if it's feasible to use `SelectorManager` abstraction, or may be it's better to create something new
Like instead of `aSocket(selectorManager)` expose something like `aSocket(socketEngine)` or something fully different.

Reply:
> _Exposing Select_ classes - not ok, as for now, it's more of an internal detail, rather then something, that could be used outside of ktor-network independent from ktor-network itself.

Actually I did not treat it as a Selector (maybe I need to rename it), but a `SocketDispatcher` or `NetworkEventLoopContext`. (I believe such a context is needed for platforms that supports running several isolated event loops, and such interfaces need commonizing)


> IMO: better to start trying to implement nodejs/mingw sockets, and see, if it's feasible to use `SelectorManager` abstraction, or may be it's better to create something new

I'll think twice about whether Interface Driven Development is suitable in this case.